lib/api/apiUtils/authorization/prepareRequestContexts.js

```diff
@@ -65,7 +65,17 @@ function prepareRequestContexts(apiMethod, request, sourceBucket,
 
     const requestContexts = [];
 
-    if (apiMethodAfterVersionCheck === 'objectCopy'
+    if (apiMethod === 'multiObjectDelete') {
+        // MultiObjectDelete does not require any authorization when evaluating
+        // the API. Instead, we authorize each object passed.
+        // But in order to get any relevant information from the authorization service
+        // for example, the account quota, we must send a request context object
+        // with no `specificResource`. We expect the result to be an implicit deny.
+        // In the API, we then ignore these authorization results, and we can use
+        // any information returned, e.g., the quota.
+        const requestContextMultiObjectDelete = generateRequestContext('objectDelete');
```

Comment:
can we instead pass `multiObjectDelete` as request context?
it will be easier to follow in lgos, and maybe eventually allow policy-based denial of multiObjectDelete...

Reply:
So, update, we cannot use this action today, or we need to add a permission, otherwise, the machinery in Arsenal will fail to convert the apiName to an action, causing issues. Also, we do not want to introduce any non-standard permission anymore.

Reply:
just to be sure, you mention:
> We expect the result to be an implicit deny

do we "expect" based on the spec (i.e. can we "garantee" we will get this), or just because this is how our implementation currently behaves (and it may change with the code) ?

i.e. I get that we don't want to introduce extra custom permissions (and agree), but I wonder if it may not be more future-proof to introduce an explicit bypass path: as it may more future-proof to introduce a dedicated and explicit case instead of relying on some corner of the implementation which may change...

Reply:
Sorry, it's not true that we always get an implicit deny, we can also have allow or explicit denies, but whatever the result is, we do not use it. We only use the authv4 internal API to get the information we need.

That is why we do not need to deny for the DeleteObjects API: in AWS, this API never returns AccessDenied, but a 200 response with an object list, and possibly AccessDenied errors, or the deletion result.

Reply:
actually, even if we ignore the result, I still have the same question: are we "garanteed" by the spec that calling the authv4 api will get us the information we need, or could it happen (with some implementation change, which may happen over time) that the information is not returned when we detect such kind of "malformed" or incomplete requests?

i.e. are we relying on a detail of the current implementation, or on a specified behavior of the contract/api ?